src/aks-preview/azext_aks_preview/tests/latest/test_natgateway.py

```diff
@@ -58,6 +58,19 @@ def test_empty_arguments(self):
         self.assertEqual(profile.managed_outbound_ip_profile.count, origin_profile.managed_outbound_ip_profile.count)
         self.assertEqual(profile.idle_timeout_in_minutes, origin_profile.idle_timeout_in_minutes)
 
+    def test_reset_empty_arguments(self):
+        origin_profile = self.nat_gateway_models.ManagedClusterNATGatewayProfile(
+            managed_outbound_ip_profile=self.nat_gateway_models.ManagedClusterManagedOutboundIPProfile(
+                count=1
+            ),
+            idle_timeout_in_minutes=4
+        )
+
+        profile = natgateway.update_nat_gateway_profile(0, None, origin_profile, models=self.nat_gateway_models)
```

Comment:
`update_nat_gateway_profile` is imported from azure-cli and the code there doesn't yet contain the fix. In addition, even if the fix there is merged into the dev branch, aks-preview's dependency on azure-cli is still at 2.49.0, which does not include fixes.

I think you need to override the implementation here in aks-preview.

Reply:
fixed. Thanks!